pkg/topology/kademlia/kademlia.go

```diff
@@ -1182,12 +1182,22 @@ func (k *Kad) EachNeighborRev(f topology.EachPeerFunc) error {
 
 // EachPeer iterates from closest bin to farthest.
 func (k *Kad) EachPeer(f topology.EachPeerFunc) error {
-	return k.connectedPeers.EachBin(f)
+	return k.connectedPeers.EachBin(k.onlyReachable(f))
 }
 
 // EachPeerRev iterates from farthest bin to closest.
 func (k *Kad) EachPeerRev(f topology.EachPeerFunc) error {
-	return k.connectedPeers.EachBinRev(f)
+	return k.connectedPeers.EachBinRev(k.onlyReachable(f))
+}
+
+// onlyReachable wraps the iterator func to filer-out unreachable peers.
+func (k *Kad) onlyReachable(f topology.EachPeerFunc) topology.EachPeerFunc {
+	return func(addr swarm.Address, po uint8) (bool, bool, error) {
+		if !k.collector.Inspect(addr).IsReachable {
+			return true, false, nil
```

Comment:
the first true stops the whole iterator, I think you just need to return false, false, nil

Reply:
correct :+1:

Reply:
Good catch, fixed!